### PR TITLE
docs:upgrade docusaurus version

### PR DESCRIPTION
**Pull request by yoonhyejin**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

Had to change some imports in Searchbar/index.js due to this change : https://github.com/facebook/docusaurus/pull/8757/files

### CODE REVIEW

File: docs-website/yarn.lock
Comment (hsheth2) [inline docs-website/yarn.lock:1]: Looks like some of the docsearch and algolia packages didn't get upgraded

Comment (yoonhyejin) [reply docs-website/yarn.lock:1]: 🤦‍♀️ my bad. just manaually updated missing packages & regenerated yarn.lock again. Could you check if this is okay?

Comment (hsheth2) [reply docs-website/yarn.lock:1]: looks good

**Review by hsheth2 — APPROVED** [review]

File: docs-website/package.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
     "markprompt": "^0.1.7",
     "react": "^18.2.0",
     "react-dom": "18.2.0",
-    "sass": "^1.43.2"
+    "sass": "^1.43.2",
+    "ts-node-dev": "^2.0.0"
```
Comment (hsheth2) [inline docs-website/package.json:43]: dont think this new dep is necessary

**Review by hsheth2 — COMMENTED** [review]
@yoonhyejin looks like the build is failing

File: docs-website/package.json (hunk 1 above)
Comment (yoonhyejin) [reply docs-website/package.json]: Yup just updated generateDocsDir for the newer version of ts-node

**Pull request merged**